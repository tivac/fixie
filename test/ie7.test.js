"use strict";

var simple = require("./util.js").simple,
    nested = require("./util.js").nested;

describe(":ie7", () => {
    it("should transform :ie7(...) into an ie7 hack (single selector)", () =>
        simple(`
            :ie7(.fooga) {
                color: red;
            }
        `)
    );

    it("should transform :ie7(...) into an ie7 hack (multiple selectors)", () =>
        simple(`
            :ie7(.fooga .booga) {
                color: red;
            }
        `)
    );

    it("should work with postcss-nested", () =>
        nested(`
            .fooga {
                color: red;
                
                :ie7(&) {
                    color: blue;
                }
            }
        `)
    );
});
