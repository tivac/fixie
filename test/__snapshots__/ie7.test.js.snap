// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`:ie7 should transform :ie7(...) into an ie7 hack (multiple selectors) 1`] = `
"*+html .fooga .booga {
    color: red;
}"
`;

exports[`:ie7 should transform :ie7(...) into an ie7 hack (single selector) 1`] = `
"*+html .fooga {
    color: red;
}"
`;

exports[`:ie7 should work with postcss-nested 1`] = `
".fooga {
    color: red;
}
    
    *+html .fooga {
        color: blue;
    }"
`;
